qutebrowser/javascript/webengine_caret.js

```diff
@@ -0,0 +1,1401 @@
+/* eslint-disable max-lines, max-len, max-statements, complexity,
+max-params, default-case, valid-jsdoc */
+/**
+ * Copyright 2018 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+ *
+ * This file is part of qutebrowser.
+ *
+ * qutebrowser is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * qutebrowser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Ported chrome-caretbrowsing extension.
+ * https://cs.chromium.org/chromium/src/ui/accessibility/extensions/caretbrowsing/
+ *
+ * Create and control div caret, which listen commands from qutebrowser,
+ * change document selection model and div caret position.
+ */
+
```

Comment:
Can you please also add the [initial comment](https://cs.chromium.org/chromium/src/ui/accessibility/extensions/caretbrowsing/caretbrowsing.js?l=5-58) from the original file? It seems to have some useful explanations.